### PR TITLE
Fix 662495

### PR DESCRIPTION
**Pull request by iainx**

### CODE REVIEW

File: main/src/core/MonoDevelop.Core/MonoDevelop.Core.Instrumentation/TimeCounter.cs [hunk 1]
```diff
@@ -141,7 +141,7 @@ public void Trace (string message)
 				}
 				traceList.TotalTime = t.Timestamp - traceList.FirstTrace.Timestamp;
 			}
-			if (counter.LogMessages) {
+			if ((bool)counter?.LogMessages) {
```
[the remark this answers is not in the document]
Comment (mrward) [reply main/src/core/MonoDevelop.Core/MonoDevelop.Core.Instrumentation/TimeCounter.cs]: Not sure this works. I just tried a console app with the following and I get an invalid operation on the if line.

```
		public static void Main (string[] args)
		{
			Foo counter = null;
			if ((bool)counter?.LogMessages) {
				Console.WriteLine ("Log messages");
			} else {
				Console.WriteLine (" Nope");
			}
		}

		class Foo
		{
			public bool LogMessages { get; set; }
		}
```

**Review by mrward — APPROVED** [review]
Code changes look fine.

Does the original null ref exception mean that this bit of code never traces anything? https://github.com/mono/monodevelop/blob/master/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor.Extension/CompletionTextEditorExtension.cs#L590

**Comment by iainx** [comment]
I'm not sure, I was trying to see how it could get to the Trace call with a null counter, but couldn't work out how it was possible.

It seems plausible that one could call `End ()` which would set counter to null and then call `Trace ()`, so checking for a null counter is some extra safety.

**Review by Therzok — APPROVED** [review]

**Comment by Therzok** [comment]
It definitely is plausible! http://source.monodevelop.com/#MonoDevelop.Ide/MonoDevelop.Ide.Editor.Extension/CompletionTextEditorExtension.cs,590

The continuation is not run on any specific scheduler, so it can run well after endtiming was done.

**Pull request merged**